ts/sagas/identification.ts

```diff
@@ -180,12 +185,28 @@ function* startAndHandleIdentificationResult(
   }
 }
 
-export function* watchIdentificationRequest(
-  pin: PinString
-): IterableIterator<Effect> {
+function* handlePinReset() {
+  // Delete the current unlock code from the Keychain
+  yield call(deletePin);
+  // invalidate the session
+  yield put(sessionInvalid());
+  // initialize the app from scratch (forcing an onboarding flow)
+  yield put(startApplicationInitialization());
+}
+
+export function* watchIdentification(pin: PinString): IterableIterator<Effect> {
+  // Watch for identification request
   yield takeLatest(
     getType(identificationRequest),
     startAndHandleIdentificationResult,
     pin
   );
+
+  // Watch for requests to reset the unlock code.
+  yield takeEvery(getType(startPinReset), handlePinReset);
```

Comment:
It looks like this action is not used anymore, do you think it is still necessary to mantain that or we can clean up the code?

It was previously only dispatched by the reset pin option on profile screen.

Reply:
No, we don't need any more.
Good catch ;)
updated in https://github.com/pagopa/io-app/pull/2195/commits/97770134f46beebcfa28a79db37ab7fb875a942c